frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/status-card/utils.ts

```diff
@@ -31,14 +31,14 @@ export const getCephHealthState: PrometheusHealthHandler = (responses = [], erro
 };
 
 export const getDataResiliencyState: PrometheusHealthHandler = (responses = [], errors = []) => {
-  if (errors[0]) {
+  const progress: number = getResiliencyProgress(responses[0]);
+  if (errors[0] || Number.isNaN(progress)) {
     return { state: HealthState.UNKNOWN };
   }
   if (!responses[0]) {
     return { state: HealthState.LOADING };
   }
```

Comment:
If responses[0] is undefined or an falsey value `progress` would be NaN and the upper `if` condition would already handle it? In what case does the HealthState go to loading?

Reply:
Right! need to move it with these changes.